lib/authentication.js

```diff
@@ -45,19 +45,30 @@ module.exports = {
             }
 
             debug("Authenticated with github: " + user.username);
-            confirmOrgMembership(user)
-            .then(function(authResponse) {
-               debug("Authenticated with org: " + user.username);
-               req.logIn(user, function(err) {
+            if (config.github.requireOrg) {
+               confirmOrgMembership(user).then(function() {
+                  debug("Authenticated with org: " + user.username);
+                  req.logIn(user, function() {
+                     debug('Got login from ' + user.username);
+
+                     // Redirect user to original URL. Fallback to '/' if DNE.
+                     const redirectUrl = req.session.auth_redirect;
+                     delete req.session.auth_redirect;
+
+                     res.redirect(redirectUrl || '/');
+                  });
```

Comment:
I wish this didn't have to be duplicated with the code below, but eh.

If you really want to, you could pull this bit out and give it a name, then reuse it in both places. Unless something cleverer occurs to you.

Reply:
Yeah I totally see that. My first attempt was to do something like

```js
let afterMembershipValidation = Promise.resolve();

if (config.github.requireOrg) {
   afterMembershipValidation = confirmOrgMembership(user);
}

afterMembershipValidation.then(() => {
   // The shared log in and redirect logic.
});
```

The only thing was the function passed to the `then` handler had a debug statement saying that the user was authenticated with the org, which only applies to one of those cases.

Maybe something like this?

```js
let afterMembershipValidation = Promise.resolve();

if (config.github.requireOrg) {
   afterMembershipValidation = confirmOrgMembership(user).then(
      () => debug('Authenticated with org: ' + user.username)
   );
}

afterMembshipValidation.then(() => {
   // The shared log in and redirect logic.
});
```

If having a no-op Promise is kind of hacky I'd be interested in making this some named function as well.

Reply:
That looks fine to me. It seems kinda equivalent to assigning 0 to a variable that you might add a number to later. If it bugs you, what about putting the `Promise.resolve()` assignment in an `else` case?